pkg/operator/status/condition.go

```diff
@@ -3,15 +3,17 @@ package status
 import (
 	"fmt"
 	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	configv1 "github.com/openshift/api/config/v1"
 	operatorv1 "github.com/openshift/api/operator/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // conditionMergeState indicates whether you want to merge all Falses or merge all Trues.  For instance, Failures merge
 // on true, but Available merges on false.  Thing of it like an anti-default.
-func unionCondition(conditionType string, defaultConditionState operatorv1.ConditionStatus, allConditions ...operatorv1.OperatorCondition) configv1.ClusterOperatorStatusCondition {
+func unionCondition(conditionType string, defaultConditionState operatorv1.ConditionStatus, gated bool, allConditions ...operatorv1.OperatorCondition) configv1.ClusterOperatorStatusCondition {
```

Comment:
instead of a flag, how about 

```go
internalUnionCondition(your ugly signature here)
unionCondition(as it was)
timeBasedUnionCondition(old signature)
```

Reply:
Done. 
~~`timeBasedUnionCondition()`~~ --> `unionInertialCondition()`